### ISSUE TITLE
A better way the extend frontmatter

### ISSUE THREAD

**Issue by Bouncey** [gatsbyjs/gatsby#2802]
I have just written a plugin that ensures all `frontmatter` have the same the same properties.

Is there a better api hook to use than `onCreateNode` seeing as I am not creating any nodes?

plugin
```js
exports.onCreateNode = async function({ node }) {
  if (node.internal.type === 'MarkdownRemark') {
    node.frontmatter = {
      ...node.frontmatter,
      challengeSeed: {
        html: false,
        css: false,
        js: false,
        jsx: false,
        ...node.frontmatter.challengeSeed
      }
    };
  }
  return node;
};
```

**Comment by KyleAMathews** [comment]
Nope! This extension point is for responding to any node creation so this exactly where you'd out validation code like this.

**Comment by Bouncey** [comment]
Sweet! Thanks for the quick reply.

I get a warning about this plugin not creating any nodes. Is there a way to make it silent?

**Comment by KyleAMathews** [comment]
Hmm that's a bug. There was a PR adding this warning recently but it shouldn't be warning based on this API. If you want to search for the PR and then add a PR removing this API that'd be great!
